src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/magical/MagicalZombiePills.java

```diff
@@ -56,11 +57,12 @@ public EntityInteractHandler getItemHandler() {
 
                 p.playSound(p.getLocation(), Sound.ENTITY_ZOMBIE_VILLAGER_CONVERTED, 1, 1);
 
-                ZombieVillager zombieVillager = (ZombieVillager) entity;
-                zombieVillager.setConversionTime(1);
+                if (entity.getType() == EntityType.ZOMBIE_VILLAGER) {
+                    healZombieVillager((ZombieVillager) entity, p);
+                }
 
-                if (SlimefunPlugin.getMinecraftVersion().isAtLeast(MinecraftVersion.MINECRAFT_1_15)) {
-                    zombieVillager.setConversionPlayer(p);
+                else if (entity.getType() == EntityType.ZOMBIFIED_PIGLIN) {
+                    healZombifiedPiglin(entity);
                 }
```

Comment:
Also, while this should not cause any probems as it is, it could be a better practice to do an `instanceof` check here instead because you are doing a class cast, just in case.